support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionBuilder.scala

```diff
@@ -5,90 +5,166 @@ import java.util.UUID
+import com.gu.support.workers.{BillingPeriod, DigitalPack}
+import com.gu.support.zuora.api.ReaderType.{Corporate, Direct, Gift}
+import com.gu.support.zuora.api.{ReaderType, SubscriptionData}
 import com.gu.zuora.subscriptionBuilders.ProductSubscriptionBuilders.{applyPromoCode, buildProductSubscription, validateRatePlan}
 import org.joda.time.LocalDate
-import com.gu.support.workers.ProductTypeRatePlans._
 
 import scala.concurrent.{ExecutionContext, Future}
 
 sealed trait SubscriptionPaymentType
 
-case class SubscriptionPaymentDirect(
-  zuoraDigitalPackConfig: ZuoraDigitalPackConfig,
+case class SubscriptionPurchase(
+  config: ZuoraDigitalPackConfig,
   maybePromoCode: Option[PromoCode],
+  billingPeriod: BillingPeriod,
   country: Country,
   promotionService: PromotionService
 ) extends SubscriptionPaymentType
 
-case class SubscriptionPaymentCorporate(
+case class SubscriptionRedemption(
   redemptionData: RedemptionData,
   getCodeStatus: GetCodeStatus
 ) extends SubscriptionPaymentType
 
 object DigitalSubscriptionBuilder {
 
+  type BuildResult = EitherT[Future, Either[PromoError, RedemptionInvalid], SubscriptionData]
+
   def build(
     digitalPack: DigitalPack,
     requestId: UUID,
     subscriptionPaymentType: SubscriptionPaymentType,
     environment: TouchPointEnvironment,
     today: () => LocalDate
-  )(implicit ec: ExecutionContext): EitherT[Future, Either[PromoError, GetCodeStatus.RedemptionInvalid], SubscriptionData] = {
+  )(implicit ec: ExecutionContext): BuildResult = {
+    val productRatePlanId = validateRatePlan(digitalPack.productRatePlan(environment, digitalPack.readerType), digitalPack.describe)
 
-    val contractEffectiveDate = today()
-    val delay = subscriptionPaymentType match {
-      case direct: SubscriptionPaymentDirect => direct.zuoraDigitalPackConfig.defaultFreeTrialPeriod + direct.zuoraDigitalPackConfig.paymentGracePeriod
-      case _: SubscriptionPaymentCorporate => 0
+    subscriptionPaymentType match {
+      case purchase: SubscriptionPurchase =>
+        buildPurchase(today(), productRatePlanId, requestId, digitalPack.readerType, purchase)
+      case redemption: SubscriptionRedemption =>
+        buildRedemption(today(), productRatePlanId, requestId, digitalPack.readerType, redemption)
     }
-    val contractAcceptanceDate = contractEffectiveDate.plusDays(delay)
+  }
 
-    val productRatePlanId = validateRatePlan(digitalPack.productRatePlan(environment, digitalPack.readerType), digitalPack.describe)
+  def buildPurchase(
+    today: LocalDate,
+    productRatePlanId: ProductRatePlanId,
+    requestId: UUID,
+    readerType: ReaderType,
+    purchase: SubscriptionPurchase
+  )(implicit ec: ExecutionContext): BuildResult = {
+
+    val (contractAcceptanceDelay, autoRenew, initialTerm) = if (readerType == Gift)
+      (0, false, purchase.billingPeriod.monthsInPeriod)
+    else
+      (purchase.config.defaultFreeTrialPeriod + purchase.config.paymentGracePeriod, true, 12)
```

Comment:
clever idea using a tuple in this way, this might make it easier to follow
```suggestion
    val (autoRenew, initialTerm, contractAcceptanceDelay) =
      if (readerType == Gift)
        (false, purchase.billingPeriod.monthsInPeriod, 0)
      else
        (true, 12, purchase.config.defaultFreeTrialPeriod + purchase.config.paymentGracePeriod)
```

Reply:
I realised later that this bit of code was moved from lower down the PR! but feel free to make changes to make it clearer as you see fit